src/Nethermind/Nethermind.Evm/ByteArrayExtensions.cs

```diff
@@ -34,19 +34,13 @@ private static ZeroPaddedSpan SliceWithZeroPadding(this ReadOnlySpan<byte> span,
             return new ZeroPaddedSpan(span.Slice(startIndex, copiedLength), length - copiedLength, padDirection);
         }
 
-        public static ZeroPaddedSpan SliceWithZeroPadding(this Span<byte> span, scoped in UInt256 startIndex, int length, PadDirection padDirection = PadDirection.Right)
-        {
-            return startIndex >= span.Length || startIndex > int.MaxValue
+        public static ZeroPaddedSpan SliceWithZeroPadding(this Span<byte> span, scoped in UInt256 startIndex, int length, PadDirection padDirection = PadDirection.Right) => startIndex >= span.Length || startIndex > int.MaxValue
                 ? new ZeroPaddedSpan(default, length, PadDirection.Right)
                 : SliceWithZeroPadding(span, (int)startIndex, length, padDirection);
-        }
 
-        public static ZeroPaddedSpan SliceWithZeroPadding(this ReadOnlySpan<byte> span, scoped in UInt256 startIndex, int length, PadDirection padDirection = PadDirection.Right)
-        {
-            return startIndex >= span.Length || startIndex > int.MaxValue
+        public static ZeroPaddedSpan SliceWithZeroPadding(this ReadOnlySpan<byte> span, scoped in UInt256 startIndex, int length, PadDirection padDirection = PadDirection.Right) => startIndex >= span.Length || startIndex > int.MaxValue
```

Comment:
duplicate?

Reply:
ReadOnlyMemory vs ReadOnlySpan or Span vs ReadOnlySpan?

Are extension methods so only show up for exact type match rather than a cast match